eng/common/pipelines/templates/steps/sparse-checkout.yml

```diff
@@ -64,6 +95,9 @@ steps:
               Write-Host "git -c advice.detachedHead=false checkout $($repository.Commitish)"
               # This will use the default branch if repo.Commitish is empty
               git -c advice.detachedHead=false checkout $($repository.Commitish)
+              if ($LASTEXITCODE) {
```

Comment:
Do we have any cases where people call sparse checkout multiple times and we fail on a try after the first and then revert back to a full clone?

Reply:
The step is idempotent so you should be able to add extra directories to an existing sparse checkout config, but the clone override is written as such that on any failure it will just pave over any existing repo files with the full clone.